Backpack-SwiftUI/CardList/Classes/BPKCardList.swift

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKCardList<Element: Identifiable, Content: View>: View {
+    private let title: String
+    private let description: String?
+    private let layout: BPKCardListLayout
+    private let initiallyShownCards: Int
+    private let elements: [Element]
+    private let cardForElement: (_ element: Element) -> Content
+    @State private var showingAllCards = false
+
+    public init(
+        title: String,
+        description: String?,
+        layout: BPKCardListLayout,
+        initiallyShownCards: Int = 3,
+        elements: [Element],
+        @ViewBuilder cardForElement: @escaping (_: Element) -> Content
+    ) {
+        self.title = title
+        self.description = description
+        self.layout = layout
+        self.initiallyShownCards = initiallyShownCards
+        self.elements = elements
+        self.cardForElement = cardForElement
+    }
+
+    public var body: some View {
+        switch layout {
+        case .rail(let sectionHeaderButton):
+            railLayout(with: sectionHeaderButton)
+        case .stack(let accessory):
+            stackLayout(with: accessory)
+        }
+    }
+
+    private func railLayout(with sectionHeaderButton: BPKCardListLayout.SectionHeaderAction?) -> some View {
+        cardListSkeleton(with: sectionHeaderButton) {
+            ScrollView(.horizontal) {
+                HStack(spacing: .base) {
+                    ForEach(elements) { index in
+                        cardForElement(index)
+                    }
+                }
+                .padding(.horizontal)
+            }
+        }
+    }
+
+    private func stackLayout(with accessory: BPKCardListLayout.Accessory?) -> some View {
+        cardListSkeleton(with: accessory?.sectionHeaderButton) {
+            let cardsShown = (showingAllCards) ? elements.count : min(initiallyShownCards, elements.count)
+            let filteredCards = elements[0..<cardsShown]
+            ForEach(filteredCards) { element in
+                cardForElement(element)
+            }
+                .padding(.horizontal)
+            accessoryView(accessory: accessory)
+                .padding(.horizontal)
+        }
+    }
+
+    private func sectionHeader(with sectionHeaderButton: BPKCardListLayout.SectionHeaderAction?) -> some View {
+        if let sectionHeaderButton {
+            return BPKSectionHeader(title: title, description: description) {
+                BPKButton(
+                    icon: sectionHeaderButton.icon,
+                    accessibilityLabel: sectionHeaderButton.accessibilityLabel,
+                    action: sectionHeaderButton.action)
+                .buttonStyle((sectionHeaderButton.style == .default) ? .primary : .primaryOnDark)
```

Comment:
Did we create a special section header button? Worried that now the button styling is here and in the SectionHeader component. 

If we ever change which styles are used, we risk having to change in two places

Reply:
That's a fair point. I'll check with design, I understood the style restriction here only applied to the button, not the whole section header, but might have misunderstood

Reply:
After discussing with design, we decided to remove this style option for now
Ideally we may want to allow any styling option for the accessory button, but went with only `.primary` for a first version